type_templates/vrm.js

```diff
@@ -39,76 +36,15 @@ const _findMaterialsObjects = (o, name) => {
   });
   return result;
 };
-const _toonShaderify = o => {
+const _toonShaderify = async o => {
   const vrmExtension = o.userData.gltfExtensions.VRM;
-  const {materialProperties} = vrmExtension;
-  
-  const materialMap = new Map();
-  const {
-    // threeTone,
-    // fiveTone,
-    twentyTone,
-  } = useGradientMapsInternal();
-  const gradientMap = twentyTone;
-  
-  for (const materialProperty of materialProperties) {
-    const {name, shader} = materialProperty;
-    if (shader === 'VRM/MToon') {
-      const objects = _findMaterialsObjects(o.scene, name);
-      for (const object of objects) {
-        const oldMaterial = object.material;
-        let newMaterial = materialMap.get(oldMaterial);
-        
-        if (!newMaterial) {
-          const opts = {};
-          const copyKeys = [
-            'alphaMap',
-            'aoMap',
-            'aoMapIntensity',
-            'bumpMap',
-            'bumpScale',
-            'color',
-            'displacementMap',
-            'displacementScale',
-            'displacementBias',
-            'emissive',
-            'emissiveMap',
-            'emissiveIntensity',
-            // 'gradientMap',
-            'lightMap',
-            'lightMapIntensity',
-            'map',
-            'normalMap',
-            'normalMapType',
-            'normalScale',
-            'wireframe',
-            'wireframeLinecap',
-            'wireframeLinejoin',
-            'wireframeLinewidth',
-            'transparent',
-            'alphaTest',
-            'opacity',
-            'side',
-            'premultipliedAlpha',
-            'polygonOffset',
-            'polygonOffsetFactor',
-            'polygonOffsetUnits',
-            'depthTest',
-            'depthWrite',
-          ];
-          for (const key of copyKeys) {
-            const value = object.material[key];
-            if (value !== undefined) {
-              opts[key] = value;
-            }
-          }
-          opts.gradientMap = gradientMap;
-          newMaterial = new THREE.MeshToonMaterial(opts);
-          materialMap.set(oldMaterial, newMaterial);
-        }
-        object.material = newMaterial;
-      }
-    }
+  const vrmImporter = new VRMMaterialImporter();
+  const materials = await vrmImporter.convertGLTFMaterials(o);

```

Comment:
As far as I can tell `convertGLTFMaterials` will already attach the converted materials.

https://github.com/pixiv/three-vrm/blob/96debe1ca3b3db8a462db0386af048f0d8339e87/packages/three-vrm/src/material/VRMMaterialImporter.ts#L99

So is the below loop even needed?